src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/GoalServiceTest.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.context.transaction.TestTransaction
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.aValidGoal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.GoalService
+
+@Deprecated("Remove this test when we have integration tests that hit controller endpoints and therefore exercise the GoalService")
+class GoalServiceTest : IntegrationTestBase() {
+
+  @Autowired
+  private lateinit var goalService: GoalService
+
+  @Test
+  @Transactional
+  fun `should create new goal given action plan does not exist`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val goal = aValidGoal()
+
+    // When
+    goalService.createGoal(goal, prisonNumber)
+
+    // Then
+    TestTransaction.flagForCommit()
+    TestTransaction.end()
+    TestTransaction.start()
+
+    val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
+    assertThat(actionPlan!!.goals).hasSize(1)
+  }
+
+  @Test
+  @Transactional
+  fun `should create new goal given action plan already exists`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+
+    val goal = aValidGoal(
+      title = "Goal 1",
+    )
+    goalService.createGoal(goal, prisonNumber)
```

Comment:
Should we commit a transaction here, before adding the additional goal?

Reply:
I'll be honest I don't fully understand the various TX boundaries, apart from to say this is needed for it to work! 😁 

The Test TX boundary is there to commit the TXs, and to allow the assertion in the `Then` block to be able to query the database directly via the repository. 

We need someone like Neil or Rob to be able to explain this properly! 🤣 , but my thoughts are that this is a temporary test anyway, so there's probably little to be gained from "rolling it in glitter" 🤣

Reply:
My assumption is that committing a transaction before starting a new one to add a Goal would reflect what would happen in the real world.

However, I forgot it's a temporary test, so I'll stand down! 😄